client.go

```diff
@@ -96,7 +96,7 @@ type ClientOpts struct {
 	Dev *bool
 
 	// Middleware is a list of middleware to apply to the client.
-	Middleware []experimental.Middleware
+	Middleware []func() middleware.Middleware
```

Comment:
Still feels weird to me to expose this to the user, especially if we add input to it further down the line. Mostly an unseen detail though I suppose with our own helpers.